src/btf.rs

```diff
@@ -120,6 +140,28 @@ impl Btf {
         Ok(types)
     }
 
+    /// Find a list of BTF types using a regex describing their name as a key.
+    #[cfg(feature = "regex")]
+    pub fn resolve_types_by_regex(&self, re: &regex::Regex) -> Result<Vec<Type>> {
```

Comment:
That sounds good but then I'm not sure how this would apply. We do have an iterator over linked types because they are like a linked list and we can do one lookup at a time. But for all the other resolving helpers it always boils down to resolving type ids, which are already stored as a vec in `BtfObj`.

So we would either:
- Hide a Vec into an iterator, which is not helping, except for maybe providing a better API?
- Rework helpers in `BtfObj` to look for the next type id in the internal vector. I'm not sure of the benefits here, it would be slower and as we already store those id in memory, by design the list has to be limited in size already.

BTW, I do like the iterator from the `btf-rs` user's point of view, it would be quite clean. But I'm really not sure it provides value at other levels.

Thoughts?